go/chat/utils/utils.go

```diff
@@ -1148,13 +1197,18 @@ func PresentConversationErrorLocal(ctx context.Context, g *globals.Context, rawC
 }
 
 func PresentConversationLocal(ctx context.Context, rawConv chat1.ConversationLocal, currentUsername string) (res chat1.InboxUIItem) {
-	var writerNames []string
-	fullNames := make(map[string]string)
+	var writerNames []chat1.UIParticipant
 	for _, p := range rawConv.Info.Participants {
-		writerNames = append(writerNames, p.Username)
-		if p.Fullname != nil {
-			fullNames[p.Username] = *p.Fullname
-		}
+		participantType := chat1.UIParticipantType_USER
+		if isPhoneOrEmail(p.Username) {
+			participantType = chat1.UIParticipantType_IMPTOFU
```

Comment:
Are you sure you want to use the name "IMPTOFU" for this? No one will no what that means in 6 months.